### PR TITLE
[GEN-539] Active session created

### PR DESCRIPTION
**Pull request by abhipi**
- Altered how sessions are created (6 hours, activated when a climb is logged)- DOCUMENTATION IN CODE
- Highlighted first item of current session.
- Session objects have not been creeated yet (Next PR)
- Shift from active -> expired session cannot be logged as sessions are temporary objects created with climbs when profile is clicked. (PUSH NOTIFICATIONS WILL NOT WORK, need to think of an alternative)

https://github.com/nagimonyc/trk-app/assets/75244191/b6b67e10-79bc-437f-b4f7-25a28d03fbf3

### CODE REVIEW

**Review by redpepper-nag — APPROVED** [review]
That looks great, well done! For the push notification, we'll brainstorm a workaround tomorrow

**Comment by abhipi** [comment]
> That looks great, well done! For the push notification, we'll brainstorm a workaround tomorrow

Yupp! Seems like the only fix is running a serverless cloud function, but doing that for all users to send push notifications is pretty ridiculous. Also cos it will need to happen pretty frequently. Still thinking over it though.....

**Comment by abhipi** [comment]
Have added a fix for Push Notifications
- When a tap is created, if there is no previous tap, a new session is created with an expiryTime
- A new session is only created after expiryTime is crossed (when new taps are made)
- When the start of a session is deleted, the next climb is marked as session start (useful for the change in sessions, better load time)
- After 6 hours of a session creation (expiryTime), push notifications are sent to the user

File: trk/functions/index.js [hunk 1]
```diff
@@ -65,4 +67,107 @@ exports.decrementUserTapCounter = functions.firestore
...
+                body: convertedPayload,
+            },
+            scheduleTime: {
+                seconds: ((data.expiryTime !== null)? data.expiryTime: (Math.floor(Date.now() / 1000) + 6 * 60 * 60)) // Scheduled  for the Session Expiration Time
+            },
+        };
+
+        await client.createTask({parent, task});
+        console.log(`Task created: ${task.name}`);
+    });
```
Comment (VGmt) [inline trk/functions/index.js:137]: nice!

File: trk/functions/index.js [hunk 2]
```diff
@@ -65,4 +67,107 @@ exports.decrementUserTapCounter = functions.firestore
...
+
+    //Notification styling and packaging
+    async function sendNotificationToUser(fcmToken, tapId, userId) {
+        const message = {
+            token: fcmToken,
+            notification: {
+                title: 'Check out your Session',
+                body: `View your recent session in Profile!`,
```
Comment (VGmt) [inline trk/functions/index.js:145]: These prompts are great, and they work for now. Note for later, we should come up with more exciting alternatives.
e.g. "Session breakdown now available!", or "Session breakdown now created!". Something like that

File: trk/src/Actions/tapActions.js [hunk 3]
```diff
@@ -12,19 +15,60 @@ export const processClimbId = (climbId, currentUser, role) => {
...
             };
-  
-            await addTap(tap);
+
+              console.log('Tap: ', tap);
+
+              if (isSessionStart) {
+                  const documentReference =  await addTap(tap);
+                  const tapDataResult = await TapsApi().getTap(documentReference.id);
+                  const scheduleFunction = firebase.functions().httpsCallable('scheduleFunction');
```
Comment (VGmt) [inline trk/src/Actions/tapActions.js:62]: that's really cool 😎 Can't wait to see it in action

File: trk/src/Screens/TabScreens/Profile/Backend/ClimberProfile.js [hunk 4]
```diff
@@ -19,14 +19,17 @@ const ClimberProfile = ({ navigation }) => {
...
+    const [currentSession, setCurrentSession] = useState([]); //to store climbs in the current session
     const [historyCount, setHistoryCount] = useState(0);
     const [sessionCount, setSessionCount] = useState(0);
     const [refreshing, setRefreshing] = useState(false);
 
 
     useFocusEffect(
         React.useCallback(() => {
-            handleTapHistory();
+            console.log('Loading Profile...'); //Loading icon on initial loads (was confusing, and seemed like data was missing otherwise)
```
Comment (VGmt) [inline trk/src/Screens/TabScreens/Profile/Backend/ClimberProfile.js:30]: ooh nice. good eye.

**Review by VGmt — COMMENTED** [review]
Looks great! Great implementations and proactiveness

**Review by VGmt — APPROVED** [review]

**Review by VGmt — APPROVED** [review]

**Review by redpepper-nag — APPROVED** [review]

**Pull request merged**